app/controllers/sessions_controller.rb

```diff
@@ -3,11 +3,16 @@ def new
   end
   
   def create
-    user = User.find_by email: login_params[:email]
+    user = User.find_by(email: login_params[:email].downcase)
     
     if user&.authenticate(login_params[:password])
-      cookies.encrypted[:user_id] = user.id
-      redirect_to root_url, notice: t('.session_created')
+      if user.activated?
+        log_in user
+        redirect_to root_url, notice: t('.session_created')
+      else
+        flash.now.alert = t('.check_activation_email')
+        render :new
```

Comment:
What happens in the following situation?

1. I register for an account
2. My activation mail is lost in the spam folder
3. I forget to check the spam folder and the activation mail is automatically deleted

Is there a way to send the activation mail again? Maybe it should be possible to send a new activation mail from this branch of code? Instead of rendering the new.html.erb template we could render an activate.html.erb template that has a link to another controller that resends the activation mail?

Reply:
That is a good idea.
but what should we do with the situation where the user closes the browser.
Meaby we could also redirect them to the activate.html.erb template when they try to login.

Reply:
If the user closes the browser and returns on a later date, they can't login if their account isn't activated right? So wouldn't they automatically see the activate.html.erb template when they try to log in?

Reply:
Yes, thats the idea.